### PR TITLE
APP-277 BWC update to Credentials v3

### PR DESCRIPTION
**Pull request by gonzaloleon**
Add "chain" to the **credential**

### CODE REVIEW

File: packages/bitcore-wallet-client/src/lib/api.ts [hunk 1]
```diff
@@ -2716,22 +2754,101 @@ export class API extends EventEmitter {
...
+    // Try to migrate to new Credentials Version
+    _.each(oldCredentials, credential => {
+      let migrated;
+
+      if (!credential.version || credential.version < 2) {
+        log.info('About to migrate : ' + credential.walletId);
+        migrated = API.upgradeCredentialsV1(credential);
+      }
+
+      if (credential.version == 2) {
```
Comment (kajoseph) [inline packages/bitcore-wallet-client/src/lib/api.ts:2835]: Need to check `migrated.version` as well

File: packages/bitcore-wallet-client/src/lib/key.ts [hunk 2]
```diff
@@ -416,6 +416,7 @@ export class Key {
 
     if (password) $.shouldBeString(password, 'provide password');
 
+    // this._checkCoin(opts.coin);
```
Comment (kajoseph) [inline packages/bitcore-wallet-client/src/lib/key.ts:419]: Is this supposed to be commented?

Comment (gonzaloleon) [reply packages/bitcore-wallet-client/src/lib/key.ts:419]: Yes, or removed. Coins (constant) contains a limited coins/token, and the goal is to add any token/chain in the future.

**Review by kajoseph — APPROVED** [review]
Looks good

**Comment by cmgustavo** [comment]
I've linked the app against this new PR and I get this error. The app doesn't start:

Screen Shot 2021-07-19 at 10 18 19

**Pull request merged**